packages/harden-plugin/src/middleware/query-complexity-plugin.ts

```diff
@@ -20,11 +20,15 @@ import { HardenPluginOptions } from '../types';
 export class QueryComplexityPlugin implements ApolloServerPlugin {
     constructor(private options: HardenPluginOptions) {}
 
-    async requestDidStart({ schema }: GraphQLRequestContext<any>): Promise<GraphQLRequestListener<any>> {
+    async requestDidStart(ctx: GraphQLRequestContext<any>): Promise<GraphQLRequestListener<any>> {
         const maxQueryComplexity = this.options.maxQueryComplexity ?? 1000;
         return {
             didResolveOperation: async ({ request, document }) => {
-                if (isAdminApi(schema)) {
+                if (this.options.skip?.(ctx)) {
```

Comment:
It would be great if this supported `Awaitable<boolean>` in addition to `boolean`. Supporting promises would allow developers to include external resources, such as making a third-party validation request.

Reply:
I purposely did not do that, because it can have severe performance impact if you do anything async in every request.

But I guess we can move that responsibility to the consumer.... Let me add it.

Reply:
And thanks for the review!